From f1c8bf461da1344ae48f456a129502f276f5fc14 Mon Sep 17 00:00:00 2001
From: Jiri Kastner <jkastner@redhat.com>
Date: Fri, 9 Oct 2015 13:50:09 +0200
Subject: [PATCH] pidstats: fix documentation indentation

Signed-off-by: Jiri Kastner <jkastner@redhat.com>
Signed-off-by: John Kacur <jkacur at redhat.com>
---
 procfs/procfs.py | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/procfs/procfs.py b/procfs/procfs.py
index 3d683c1ec5c6..0706fbc3debd 100755
--- a/procfs/procfs.py
+++ b/procfs/procfs.py
@@ -412,20 +412,20 @@ class pidstats:
 		return key in self.processes
 
 	def reload(self):
-	"""
-	This operation will trow away the current dictionary contents, if any, and
-	read all the pid files from /proc/, instantiating a 'process' instance for
-	each of them.
+		"""
+		This operation will throw away the current dictionary contents, if any, and
+		read all the pid files from /proc/, instantiating a 'process' instance for
+		each of them.
 
-	This is a high overhead operation, and should be avoided if the perf python
-	binding can be used to detect when new threads appear and existing ones
-	terminate.
+		This is a high overhead operation, and should be avoided if the perf python
+		binding can be used to detect when new threads appear and existing ones
+		terminate.
 
-	In RHEL it is found in the python-perf rpm package.
+		In RHEL it is found in the python-perf rpm package.
 
-	More information about the perf facilities can be found in the 'perf_event_open'
-	man page.
-	"""
+		More information about the perf facilities can be found in the 'perf_event_open'
+		man page.
+		"""
 		del self.processes
 		self.processes = {}
 		pids = os.listdir(self.basedir)
-- 
2.4.3

